packages/system/src/system.ts

```diff
@@ -40,16 +46,31 @@ interface GetStyleObject {
  */
 export const toCSSObject: GetStyleObject = ({ baseStyle }) => (props) => {
   const { theme, css: cssProp, __css, sx, ...rest } = props
-  const styleProps = objectFilter(rest, (_, prop) => isStyleProp(prop))
-  const finalBaseStyle = runIfFn(baseStyle, props)
-  const finalStyles = Object.assign(
-    {},
-    __css,
-    finalBaseStyle,
-    filterUndefined(styleProps),
-    sx,
+  const stylePrecedence = []
+
+  // Styles are applied by precedence (lowest to highest)
+  //   __css
+  //   baseStyle
+  //   other (apply, layerStyles, textStyles)
+  //   styleProps
+  //   sx
+  //   css
+  stylePrecedence.push({ ...__css })
+  stylePrecedence.push(Object.assign({}, runIfFn(baseStyle, props)))
+  stylePrecedence.push(objectFilter(rest, (_, prop) => isOtherProp(prop)))
+
+  const styleProps = objectFilter(
+    rest,
```

Comment:
As we discussed during the call, let's see how we can do both `objectFilter` in a single call. If need be, we can write a custom `objectFilter`. Here's what a sample implementation can look like

```jsx
const { styleProp, otherProp, domProps } = splitProps(rest)
```